airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -0,0 +1,461 @@
+        self,
+        verbose: Optional[bool] = False,
+    ) -> List:
+        """
+        Lists all Amazon EKS Clusters in your AWS account.
+
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List containing the cluster names.
+        :rtype: List
+        """
+        eks_client = self.conn
+        api_call = partial(eks_client.list_clusters)
+
+        return self._list_all(api_call=api_call, response_key="clusters", verbose=verbose)
+
+    def list_nodegroups(
+        self,
+        clusterName: str,
+        verbose: Optional[bool] = False,
+    ) -> List:
+        """
+        Lists all Amazon EKS Nodegroups associated with the specified cluster.
+
+        :param clusterName: The name of the Amazon EKS Cluster containing nodegroups to list.
+        :type clusterName: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of nodegroup names within the given cluster.
+        :rtype: List
+        """
+        eks_client = self.conn
+        api_call = partial(eks_client.list_nodegroups, clusterName=clusterName)
+
+        return self._list_all(api_call=api_call, response_key="nodegroups", verbose=verbose)
+
+    def _list_all(self, api_call: Callable, response_key: str, verbose: bool) -> List:
+        """
+        Repeatedly calls a provided boto3 API Callable and collates the responses into a List.
+
+        :param api_call: The api command to execute.
+        :type api_call: Callable
+        :param response_key: Which dict key to collect into the final list.
+        :type response_key: str
+        :param verbose: Provides additional logging if set to True.  Defaults to False.
+        :type verbose: bool
+
+        :return: A List of the combined results of the provided API call.
+        :rtype: List
+        """
+        name_collection = []
+        token = DEFAULT_PAGINATION_TOKEN
+
+        try:
+            while token != "null":
+                response = api_call(nextToken=token)
+                # If response list is not empty, append it to the running list.
+                name_collection += filter(None, response.get(response_key))
+                token = response.get("nextToken")
+
+            self.log.info("Retrieved list of %s %s.", len(name_collection), response_key)
+            if verbose:
+                self.log.info("%s found: %s", response_key.title(), name_collection)
+
+        except ClientError as e:
+            self.log.error(e.response["Error"]["Message"])
+            raise e
+        return name_collection
+
+
+def generate_config_file(
+    eks_cluster_name: str,
+    eks_namespace_name: str,
+    pod_name: str,
+    aws_profile: Optional[str],
+    pod_username: Optional[str] = DEFAULT_POD_USERNAME,
+    pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
+    aws_region: Optional[str] = None,
+) -> str:
```

Comment:
```suggestion
def generate_config_file(
    eks_cluster_name: str,
    eks_namespace_name: str,
    pod_name: str,
    pod_username: Optional[str] = DEFAULT_POD_USERNAME,
    pod_context: Optional[str] = DEFAULT_CONTEXT_NAME,
    aws_conn_id: str = 'aws_default'
) -> str:
```

Reply:
We should accept conn_id to ensure unify credential managementt.

Reply:
Should be addressed by https://github.com/apache/airflow/pull/16571/commits/8687b736994d1911011dd8bd2b2903d82ad30a8c